Teapot/MockTeapot.swift

```diff
@@ -95,4 +117,20 @@ open class MockTeapot: Teapot {
             completion(nil, TeapotError.missingMockFile(resource))
         }
     }
+
+    private func checkHeadersAgainstExpected(headers: [String: String]?) -> Bool {
+        guard !headersToCheckFor.isEmpty else { /* nothing to check */ return true }
+
+        guard let received = headers else { /* We want to check but there's nothing to check */ return false }
+
+        for (key, value) in headersToCheckFor {
```

Comment:
Can't there be a function that just checks wether `receivedHeaders.containts(headersToCheckFor)`?

Reply:
There's only a `contains(where:)` on dictionaries - I don't think it'd actually be that much shorter than this, and it probably wouldn't be as clear